src/libcommon/log/sentry/sentryhandler.cpp

```diff
@@ -115,10 +112,17 @@ void SentryHandler::captureMessage(SentryLevel level, const std::string &title,
 
     updateEffectiveSentryUser(user);
 
-    sentry_capture_event(
-        sentry_value_new_message_event(static_cast<sentry_level_t>(event.level), event.title.c_str(), event.message.c_str()));
+    sendEventToSentry(event.level, event.title, event.message);
 }
 
+void SentryHandler::sendEventToSentry(const SentryLevel level, const std::string &title, const std::string &message) const {
+    sentry_capture_event(sentry_value_new_message_event(static_cast<sentry_level_t>(level), title.c_str(), message.c_str()));
+}
+
+SentryHandler::SentryHandler(int maxCaptureCountBeforeRateLimit, int minUploadIntervalOnRateLimit) :
```

Comment:
Because this constructor is only used in unit tests, I would define it in the mock class instead. (This improves intent clarity; this way no one is tempted to use it outside of this context).

Reply:
> Doing this means _maxCaptureCountBeforeRateLimit and minUploadIntervalOnRateLimit will need to be protected members, which will trigger Sonar. 

I didn't know this would upset Sonar. My feeling is that allowing data members to be only `protected` is less confusing or noisy on the implementation side than a constructor dedicated to unit tests only. What would Sonar Cloud think about making the test class class friend?

Reply:
Here are the details of why Sonar considers this a critical maintainability issue: https://rules.sonarsource.com/cpp/RSPEC-3656/?search=protected. 
Not tested yet, but I don't think Sonar would flag this, as it is already the case for some other classes (like `SyncPal`, for example). 

However, regardless of what I mentioned in my previous comment 👀, I am unsure whether it's better to use a `friend` declaration or to add a protected setter. It doesn't seem unreasonable to me to allow a derived class to modify these settings of the base class.

Reply:
Here it is 👍🏾: [2d8d492](https://github.com/Infomaniak/desktop-kDrive/pull/291/commits/2d8d492c2daa421185a7a41775c4b5b8a44d8b47)

PS: 
> looks better than the public alternative constructor

In the previous implementation the test specific constructor was protected

Reply:
Oh, yes, sorry. What troubled me most was the visibility in the base class of a constructor that should be used only in unit tests. Sorry if I wasn't clear.